src/app/app-routing.module.ts

```diff
@@ -14,6 +14,20 @@ const routes: Routes = [
     path: 'counters',
     loadChildren: './counters/counters.module#CountersPageModule'
   },
+  {
+    path: 'counter-analytics/:id',
+    resolve: {
+      counter: CounterRepositoryService,
+    },
+    loadChildren: './counter-analytics/counter-analytics.module#CounterAnalyticsPageModule'
+  },
+  {
+    path: 'counter-analytics/:id/day-of-week-histogram',
```

Comment:
I would not have created a component specifically for this visualization. But a generic component which contains the container and takes as arguments: 

* the counter id 
* the visualization to render

so in terms of routing configuration:
`counter-analytics/:id/:visualization-name`
or maybe
`counter-analytics/:visualization-name/:id`

the visualizations themselves could just be classes which share the same `draw(chart-container)` interface. And are loaded dynamically based on the visualization parameter by this component.

Reply:
I'm not sure. We create a new class per visualization in both approaches, which is a good thing. But if we generalize the page we loose the freedom to define content which is specifically for one visualization (descriptions, controls, special CSS, ...).

Reply:
Hmm.. I understand that this gives us bigger flexibility when building new visualizations. On the other hand if every visualization has such flexibility we might create visualizations which are completely different in their look & feel from all the other ones. Which might create bad user experience.

Reply:
I propose to implement a few more visualizations using the current approach and then we can reconsider.